src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -391,8 +403,8 @@ protected function getFormActions()
         $majorActions->setFieldHolderTemplate(get_class($majorActions) . '_holder_buttongroup');
         $actions->push($majorActions);
 
-        if ($this->record->ID !== 0) { // existing record
-            if ($this->record->canEdit()) {
+        if ($this->record->ID !== null && $this->record->ID !== 0) { // existing record
```

Comment:
In this class the only methods we expect which _aren't_ on `ViewableData` are either defined in the `DataObjectInterface` or have ways to pass the information in via a setter method on the component. None of those setter methods accept a callback, which would be necessary for this scenario. I don't think this is a case where we need to enforce a specific method or callback, since we know imperically that if the ID is missing, it's not in the DB.

What's more, the point of this is to explicitly allow data that _isn't_ in the database. `isInDB` tells you if the record is in the database. The sensible return value would always be `false` for the types of things we want to handle here.